### ISSUE TITLE
[Feature request] Deployment on Microsoft Store

### ISSUE THREAD

**Issue by AntonKrug** [pbek/QOwnNotes#2740]
This is not an issue, it's a feature request.
Would you consider deploying the application on Microsoft Store?

**Comment by pbek** [comment]
If there is a way to deploy the **content a zip file** and there is a way to script that, sure. 😁

**Comment by pbek** [comment]
winget for example doesn't allows to deploy the content a zip file...

**Comment by AntonKrug** [comment]
Would it require creating an MSI package first?

https://github.com/jpakkane/msicreator

**Comment by pbek** [comment]
For sure (which I don't want to support)... And then it surely needs to be signed, I assume offline and on Windows (which I don't use in my offline build pipeline and then uploaded to a store (most likely manually).

**Comment by pbek** [comment]
If anyone wants to attempt to build such a pipeline, I'd be happy to assist.

**Comment by pbek** [comment]
Maybe getting something to work like a NSI installer would be a good starting point (I don't know if MS supports that):

https://github.com/Serial-Studio/Serial-Studio/blob/0a16c90acdde2e99a5e78dc158eae38cf0926c19/.github/workflows/deploy.yml#L201-L204

**Comment by Kroc** [comment]
Winget *does* support deploying a zip file; it was added at the start of the year: https://devblogs.microsoft.com/commandline/windows-package-manager-1-4/

**Comment by pbek** [comment]
In https://devblogs.microsoft.com/commandline/windows-package-manager-1-4/#zip-archive I can only see that it supports executing installers from a ZIP archive... 🤔 

Maybe someone has a proof of concept or an application in the Microsoft Store that just extracts a ZIP file?

**Comment by Kroc** [comment]
I believe you need to use "zip" as the package type, and "portable" as the nested type; take for example: https://github.com/microsoft/winget-pkgs/blob/master/manifests/n/NirSoft/NirCmd/2.86/NirSoft.NirCmd.installer.yaml

**Comment by pbek** [comment]
If there is a documentation about the installer format and how to get it in the store and a way to automate updates in GitHub CI I might consider it...

**Comment by Kroc** [comment]
Packages are added via PRs to the winget package repository: https://github.com/microsoft/winget-pkgs/ ; full instructions are provided there!

**Comment by pbek** [comment]
Thank you! If anyone wants to play around with it, I would certainly assist. But I currently have almost no access to a Windows box to test.

**Comment by chris-mosley** [comment]
@pbek I'm willing to go through the headache of learning and getting this into the winget repo if you're ok with that?

**Comment by pbek** [comment]
Yes, thank you very much, @chris-mosley! It would be particularly beneficial to know if it's something that can be integrated into the current GitHub build pipeline.

**Comment by chris-mosley** [comment]
It looks like it should be pretty straightforward.  Just generating some YAML manifest files and opening a PR with the package repo.  I started messing with it last night, I'm hoping we can have just a single multi language manifest but first blush appears that we'll need to generate one for each language and have a bunch of nearly identical manifests pointing to the same file ¯\\_(ツ)_/¯

**Comment by pbek** [comment]
Thank you for the update. Sounds great so far. 👍🏻

**Comment by chris-mosley** [comment]
I have this PR open to get version 23.12.3 into the repo.  I believe it would only be listed on systems with the en_US local for now.  If this is a success and they don't bounce it back I'll go through the trouble of figuring out all the 60+ locales ill need to add -_-

https://github.com/microsoft/winget-pkgs/pull/130310

**Comment by pbek** [comment]
Ah, nice! Thank you! Doing that for all languages will be hard, I guess...
And I wonder how that all can be automated as part of the QOwnNotes build pipeline...

**Comment by pbek** [comment]
I wonder if there are bots, like for flatpak or https://github.com/r-ryantm for nix, that are detecting new releases and creating such pull requests on their own.

**Comment by chris-mosley** [comment]
I don't think the languages will be terribly difficult to add just as slugs for now.  I think you could literally copy the
pbek.QOwnNotes.locale.en-US.yaml  yaml file and find/replace en_US with en_AT or whatever.  

The only complication IMO is that ideally we would like the tags/search keywords to be in the language of that particular manifest.  ie I added 
Tags:
- notes
- notetaking
- qownnotes
- todo
- markdown

We would just need to have equivalents for every language which is just a bit tedious to do all in one shot lol.

I'm hoping that we can do a "Universal" package but I'm not terribly optimistic.

re: bots like flatpack.  I think it should be pretty easy to do.  I think you just have to take the manifest I pushed and update the version, release notes etc and the installerSha256 (which I think is just the hash of the zip file).

You may also want to check the manifests I pushed to make sure you're happy with all the information I gave, I put my best guesses in for the different fields, if there is anything you disagree with let me know and I can update the PR.

**Comment by pbek** [comment]
As long we don't need to translate (and support) everything in all languages, I'm fine with it. 😅 

Ad bots: I wonder if there are already such bots out there. For example, the one for Nix is using https://repology.org/ as data source.
Ad manifest: Where did you get the `QOwnNotes is the open source notepad with markdown support and todo list manager`? Did I still write `Markdown` in lowercase somewhere (I saw it written wrong in the README, for example 🙈)? Maybe something like `QOwnNotes is a plain-text file notepad and todo-list manager with Markdown support and Nextcloud / ownCloud integration.` would be best.

**Comment by pbek** [comment]
I tried to play around with Komac, but ran into https://github.com/russellbanks/Komac/issues/480. 😬

**Comment by o-l-a-v** [comment]
Some changes since last comment:

* 2025-09-10 "Free developer registration for individual developers on Microsoft Store": <https://blogs.windows.com/windowsdeveloper/2025/09/10/free-developer-registration-for-individual-developers-on-microsoft-store/>
* From <https://github.com/russellbanks/Komac/issues/480>
  * 2025-10-17 comment <https://github.com/russellbanks/Komac/issues/480#issuecomment-3416329247>
  > I found that adding workflow to the token was sufficient to get past this error.

**Comment by pbek** [comment]
Isn't there https://winget.ragerworks.com/package/pbek.QOwnNotes?

**Comment by o-l-a-v** [comment]
WinGet != Microsoft Store.

**Comment by pbek** [comment]
I thought they are using the same store. They are both from Microsoft 🤔

**Comment by o-l-a-v** [comment]
WinGet only stores info on where to download the artifacts and how to install them.

Microsoft Store is more like Apple Store and Google Store on Android.

WinGet can also install apps from Microsoft Store.

**Comment by pbek** [comment]
Thank you.
